api/cleanup_unused_api_indexes.py

```diff
@@ -0,0 +1,189 @@
+#!/usr/bin/env python
+"""
+This script deletes unused indexes in the API cluster.
+
+This reduces disk usage, and in our experience reduces CPU and memory pressure in the cluster.
+
+This script will go through all the indexes in the cluster.  If it thinks an index is safe
+to delete (i.e. it's not being used in the API and is older than the API indexes), it will
+ask you to confirm you want to delete it.
+"""
+
+import click
+import json
+
+import boto3
+import httpx
+import humanize
+
+
+def get_session_with_role(role_arn):
+    """
+    Returns a boto3.Session that uses the given role ARN.
+    """
+    sts_client = boto3.client("sts")
+
+    assumed_role_object = sts_client.assume_role(
+        RoleArn=role_arn, RoleSessionName="AssumeRoleSession1"
+    )
+    credentials = assumed_role_object["Credentials"]
+    return boto3.Session(
+        aws_access_key_id=credentials["AccessKeyId"],
+        aws_secret_access_key=credentials["SecretAccessKey"],
+        aws_session_token=credentials["SessionToken"],
+    )
+
+
+def get_api_es_client(session):
+    """
+    Returns an Elasticsearch client for the catalogue cluster.
+    """
+    secrets = session.client("secretsmanager")
+
+    credentials = json.loads(
+        secrets.get_secret_value(SecretId="elasticsearch/api_cleanup/credentials")[
+            "SecretString"
+        ]
+    )
+
+    return httpx.Client(
+        base_url=credentials["endpoint"],
+        auth=(credentials["username"], credentials["password"]),
+    )
+
+
+def list_indexes(es_client):
+    """
+    Returns a list of indexes in the Elasticsearch cluster, sorted by index name.
+    """
+    resp = es_client.get("/_cat/indices", params={"format": "json"})
```

Comment:
To avoid some of the index name checking you can use
```suggestion
    resp = es_client.get("/_cat/indices/works-*,images-*", params={"format": "json"})
```

[See here for more details](https://www.elastic.co/guide/en/elasticsearch/reference/current/cat-indices.html#cat-indices-api-path-params)

Reply:
Nice!

I'm going to leave the name checking as-is because it's another safety check, and in this sort of script a bit of extra checking never goes amiss.